### PR TITLE
Adding Weather Cog

### PR DESCRIPTION
**Pull request by TechDufus**
This PR for the Weather cog should operate identically to the original weather command.

```
.weather 60110
```
This will generate a weather status for this zip code.

I would like @xdega to confirm that I've imported the API key in the correct location in this cog class.

### CODE REVIEW

File: cogs/weather.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import discord
+from discord.ext import commands
+
+from random import randint
```
Comment (TechDufus) [inline cogs/weather.py]: This is not needed.

File: cogs/weather.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import discord
+from discord.ext import commands
+
+class Roll(commands.Cog):
```
Comment (xdega) [inline cogs/weather.py]: The class name here should be something that describes the command, or collection of commands. 
In this case, probably something like "Weather"? 

At the very least, let's not overwrite the Roll command. 😄 
```suggestion
class Weather(commands.Cog):
```

File: cogs/weather.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                7: ':foggy: :eyeglasses:',
+                8: ':white_sun_small_cloud: :sunny: :fire:'
+            }
+            emojistr = emojistr.get(code, '')
+        else:
+            emojistr = ':cloud: :slight_frown: :cloud: :slight_frown: :cloud:'
+        await ctx.send(f"**The Weather for {station} is:** {forecast} {emojistr}")
+
+def setup(client):
+    client.add_cog(Roll(client))
```
Comment (xdega) [inline cogs/weather.py]: The add_cog() function expects an instance of the class defined above. If you went with the previous suggestion, that would be the following: 
```suggestion
    client.add_cog(Weather(client))
```

File: cogs/weather.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+        if int(code) <= 800:
+            code = int(str(code)[:1])
+            emojistr = {
+                2: ':thunder_cloud_rain: :fearful:',
+                3: ':white_sun_rain_cloud:',
+                5: ':cloud_rain: :frowning:',
+                6: ':cloud_snow: :snowflake: :snowman:',
+                7: ':foggy: :eyeglasses:',
+                8: ':white_sun_small_cloud: :sunny: :fire:'
```
Comment (xdega) [inline cogs/weather.py]: Didn't we decide to just use double-quotes across the board?

File: cogs/weather.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import discord
+from discord.ext import commands
+
```
Comment (xdega) [inline cogs/weather.py:4]: According to the default linting rules, we need an extra newline here:
```suggestion


```

File: cogs/weather.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+                5: ':cloud_rain: :frowning:',
+                6: ':cloud_snow: :snowflake: :snowman:',
+                7: ':foggy: :eyeglasses:',
+                8: ':white_sun_small_cloud: :sunny: :fire:'
+            }
+            emojistr = emojistr.get(code, '')
+        else:
+            emojistr = ':cloud: :slight_frown: :cloud: :slight_frown: :cloud:'
+        await ctx.send(f"**The Weather for {station} is:** {forecast} {emojistr}")
+
```
Comment (xdega) [inline cogs/weather.py:50]: According to default linting rules, we need an extra newline here: 
```suggestion


```

**Review by xdega — CHANGES_REQUESTED** [review]
I know we haven't talked about using a formatter, but it is worth noting that I am using `autopep8` to ensure some baseline structure is followed in the code formatting.

https://code.visualstudio.com/docs/python/editing#_formatting

File: cogs/weather.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import discord
```
Comment (xdega) [inline cogs/weather.py:2]: Don't forget to import the `os` library, if we want to use environment variables: 
```suggestion
import os
import discord
```

File: cogs/weather.py (hunk 4 above)
Comment (TechDufus) [reply cogs/weather.py]: I'm seeing more and more examples of people using double quotes even for non-evaluated strings. Let's go with this going forward.

File: cogs/weather.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    WEATHER_API_KEY = os.environ["open_weather_map_key"]
+
+    def __init__(self, client):
+        self.client = client
+        print("Loading Command: Weather")
+
+    # Events
+    @commands.Cog.listener()
+    async def on_ready(self):
+        print("Bot is Ready")
```
Comment (xdega) [inline cogs/weather.py:17]: Sorry I didn't catch this before, but this not actually needed here in this cog (nor is it really required in the Roll command cog). It does, however, illustrate how you can make a bot do things based on a certain event. This is how you would write a custom "greeting", for example.

Let's not worry about it right now. I want this to move forward. But lets just keep in mind that we may want to remove these pointless events in the future, but keep in mind that events can be a useful tool for future functionality.

**Review by xdega — APPROVED** [review]

**Pull request merged**